File: apps/driver-android/src/main/kotlin/lk/mageride/driver/onboarding/FeatureSlides.kt
package lk.mageride.driver.onboarding

import androidx.annotation.StringRes
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.AccountBalanceWallet
import androidx.compose.material.icons.outlined.Bolt
import androidx.compose.material.icons.outlined.Campaign
import androidx.compose.ui.graphics.vector.ImageVector
import lk.mageride.driver.R

/**
 * One slide of SCR-DA-002's feature carousel.
 *
 * @property title The slide's headline.
 * @property body One sentence under it.
 * @property caption The illustration panel's own label.
 * @property icon Stands in for the illustration — see [FeatureSlides].
 */
internal data class FeatureSlide(
    @param:StringRes val title: Int,
    @param:StringRes val body: Int,
    @param:StringRes val caption: Int,
    val icon: ImageVector,
)

/**
 * AL-28's **3-slide feature infographic**, mirroring the passenger app's SCR-PA-002.
 *
 * BR-25.1: *"a 3-slide feature-infographic carousel (content-svc strings, Si/Ta/En) above the
 * language & city selectors. Presentation only; no gating."* The four features it has to land
 * before a driver picks a language are the ones the wireframe's own slide-1 copy names —
 * onboarding, 15-second dispatch, Directional Travel, and the in-app wallet and daily fee.
 *
 * ### Why the strings are resources rather than content-svc's
 *
 * AL-28 says content-svc serves them, and `backend/contracts/content.yaml` declares no route that
 * does: it carries `GET /v1/config/cities`, the mTLS notification templates and the in-app
 * broadcasts, and nothing else. Shipping the copy as trilingual resources satisfies the *rule*
 * that matters — every string exists in Si, Ta and En, and `StringResourceTest` enforces it — and
 * leaves one list to replace when a route lands. Raised as a micro-change-set in the C068 handoff.
 */
internal object FeatureSlides {

    /** The three slides, in order. The dots and the pager both count from this list. */
    val All: List<FeatureSlide> = listOf(
        FeatureSlide(
            title = R.string.onboarding_slide_earn_title,
            body = R.string.onboarding_slide_earn_body,
            caption = R.string.onboarding_slide_earn_caption,
            icon = Icons.Outlined.Campaign,
        ),
        FeatureSlide(
            title = R.string.onboarding_slide_dispatch_title,
            body = R.string.onboarding_slide_dispatch_body,
            caption = R.string.onboarding_slide_dispatch_caption,
            icon = Icons.Outlined.Bolt,
        ),
        FeatureSlide(
            title = R.string.onboarding_slide_wallet_title,
            body = R.string.onboarding_slide_wallet_body,
            caption = R.string.onboarding_slide_wallet_caption,
            icon = Icons.Outlined.AccountBalanceWallet,
        ),
    )
}

File: apps/driver-ios/DriverApp/Onboarding/FeatureSlides.swift
import Foundation

/// One slide of SCR-DI-002's feature carousel.
///
/// - Parameters:
///   - titleKey: The slide's headline.
///   - bodyKey: One sentence under it.
///   - captionKey: The illustration panel's own label.
///   - symbolName: Stands in for the illustration — see ``FeatureSlides``.
struct FeatureSlide: Identifiable {
    let titleKey: String
    let bodyKey: String
    let captionKey: String
    let symbolName: String

    var id: String { titleKey }
}

/// AL-28's **3-slide feature infographic**, mirroring the passenger app's SCR-PA-002.
///
/// BR-25.1: *"a 3-slide feature-infographic carousel (content-svc strings, Si/Ta/En) above the
/// language & city selectors. Presentation only; no gating."* The features it has to land before a
/// driver picks a language are the ones the wireframe's own slide-1 copy names — onboarding,
/// 15-second dispatch, Directional Travel, and the in-app wallet and daily fee.
///
/// ### Why the strings are resources rather than content-svc's
///
/// The same three keys, the same three sentences and the same three icons as
/// `apps/driver-android/.../onboarding/FeatureSlides.kt`, and for a reason that is now a *parity*
/// reason rather than a contract one. C068 shipped bundled copy because `content.yaml` declared no
/// route that served it; **MCS-03 has since added `GET /v1/content/onboarding/{audience}`** and
/// C076's passenger app consumes it. Moving one driver app onto the route and not the other would
/// make the two carousels differ on first run, which is exactly what this component's parity fence
/// forbids — so both move together, in a paired change. Recorded in the C086 handoff.
///
/// **The illustration is an SF Symbol, and that is the same dependency wall C076 hit.**
/// `OnboardingSlide.illustrationRef` is *"an app-bundled asset key, or an absolute https URL"* —
/// content-svc serves the reference and never image bytes — and this target ships no artwork and no
/// remote image loader. The per-slot symbol is the honest stand-in.
enum FeatureSlides {

    /// The three slides, in order. The dots and the pager both count from this list.
    static let all: [FeatureSlide] = [
        FeatureSlide(
            titleKey: "onboarding_slide_earn_title",
            bodyKey: "onboarding_slide_earn_body",
            captionKey: "onboarding_slide_earn_caption",
            symbolName: "megaphone"
        ),
        FeatureSlide(
            titleKey: "onboarding_slide_dispatch_title",
            bodyKey: "onboarding_slide_dispatch_body",
            captionKey: "onboarding_slide_dispatch_caption",
            symbolName: "bolt"
        ),
        FeatureSlide(
            titleKey: "onboarding_slide_wallet_title",
            bodyKey: "onboarding_slide_wallet_body",
            captionKey: "onboarding_slide_wallet_caption",
            symbolName: "wallet.pass"
        ),
    ]
}
